daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -2364,6 +2361,35 @@ sealed abstract class DocumentPart(part: Node, parent: Document) {
     }
   }.trim.toUpperCase()
 
+  /*
+  * Description: Compare ByteDocumentPart/BitsDocumentPart string to a string of valid characters.
```

Comment:
Instead of describing your algorithm here, imagine the user doesn't care about the algorithm they just need to know how to call it properly. I don't think you need to say more than "removes comments from '//' to end of line or end of document part element. Any characters other than those explicitly allowed are also considered comments and are removed."